### PR TITLE
NMS-13456: Collectors edits from MMahacek

### PR DESCRIPTION
**Pull request by mmahacek**
### All Contributors

* [X] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?

This PR is review/edits to https://github.com/OpenNMS/opennms/pull/3553

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13456

### CODE REVIEW

File: docs/modules/operation/pages/performance-data-collection/collectors/snmp/SnmpCollector.adoc [hunk 1]
```diff
@@ -56,170 +59,173 @@ Define SNMP collection in `etc/datacollection-config.xml` and `etc/datacollectio
...
-
-*Scenario*
-
-a) Node A belongs to categories CatA and CatB, and has a sysObjectID of .1.100.1.1
-
-b) Node B belongs to category CatB, and has a sysObjectID of .1.100.1.2
+=== SNMP data collection example
 
-Be aware of the distinction between the following concepts:
+The following scenario steps through creating a new datacollection definition and assumes the following facts:
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/collectors/snmp/SnmpCollector.adoc:79]:
```suggestion
The following scenario steps through creating a new datacollection definition and assumes the following:
```

File: docs/modules/operation/pages/performance-data-collection/collectors/snmp/SnmpCollector.adoc [hunk 2]
```diff
@@ -56,170 +59,173 @@ Define SNMP collection in `etc/datacollection-config.xml` and `etc/datacollectio
...
 
-. Build the effective snmp-collection based on the systemDef list associated with a given list of datacollection-groups.
-. Build the effective list of mibObj to collect from a given node using its sysObjectID.
+* Node A belongs to categories CatA and CatB, and has a sysObjectID of `.1.100.1.1`.
+* Node B belongs to category CatB, and has a sysObjectID of `.1.100.1.2`.
 
-The first process starts when the Spring context initializes.
-It also happens during runtime when a change in `datacollection-config.xml` is detected (not the files in the `datacollection` directory).
+If you want to collect different OIDs based on a categories filter, this requires you to define separate Collectd packages.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/collectors/snmp/SnmpCollector.adoc:84]:
```suggestion
If you want to collect different OIDs based on a categories filter, you must define separate Collectd packages.
```

File: docs/modules/operation/pages/performance-data-collection/collectors/snmp/SnmpCollector.adoc [hunk 3]
```diff
@@ -56,170 +59,173 @@ Define SNMP collection in `etc/datacollection-config.xml` and `etc/datacollectio
...
-Now, as the node’s sysObjectID is .1.100.1.1, only the systemDefs Collect-A, Collect-C, Collect-D, Collect-F, and Collect-G will be included, as those are the only ones with a sysoid or sysoidMask that matches the sysObjectID.
+Because this node matches two collectd packages for the SNMP service, the collector uses both collections (group1 and group2) and will check Collect-A through Collect-I.
+Even though UniqueB is referenced twice, it will only be included once.
 
-However, it will add the mibObj groups in the order they appear.
-If one systemDef references a group already included, it won’t add it again (knowing the groups are done by checking their names).
+Since the node’s sysObjectID is .1.100.1.1, only the systemDefs Collect-A, Collect-C, Collect-D, Collect-F, and Collect-G will be included, as those are the only ones with a sysoid or sysoidMask that matches the sysObjectID.
 
-The idea is to extract the list of systemDefs that match the sysObjectID, to get the list of mibObj groups, to finally get the list of OIDs to retrieve via SNMP.
+The mibObj groups will be added in the order they appear.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/collectors/snmp/SnmpCollector.adoc:215]:
```suggestion
The mibObj groups are added in the order they appear.
```

File: docs/modules/operation/pages/performance-data-collection/collectors/snmp/SnmpCollector.adoc [hunk 4]
```diff
@@ -56,170 +59,173 @@ Define SNMP collection in `etc/datacollection-config.xml` and `etc/datacollectio
...
+If one systemDef references a group name that already included, it won’t be added again.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/collectors/snmp/SnmpCollector.adoc:216]:
```suggestion
If one systemDef references a group name that is already included, it won’t be added again.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]

File: docs/modules/operation/pages/performance-data-collection/collectors/xml.adoc [hunk 5]
```diff
@@ -74,12 +75,12 @@ The request element can have the following optional child elements:
 [options="header"]
 [cols="1,3,2"]
 |===
-| Parameter               | Description                                                           | Default
-| timeout               | The connection and socket timeout in milliseconds.                     | n/a
-| retries               | How often to repeat the request in case of an error.         | 0
+| Parameter             | Description                                                                          | Default
+| timeout               | The connection and socket timeout in milliseconds.                                   | n/a
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/collectors/xml.adoc:79]:
```suggestion
| timeout               | The connection and socket timeout in milliseconds                                   | n/a
```

File: docs/modules/operation/pages/performance-data-collection/collectors/jmx.adoc [hunk 6]
```diff
@@ -11,6 +11,7 @@ Attributes are extracted from the available MBeans.
 | Class Name          | `org.opennms.netmgt.collectd.Jsr160Collector`
 | Package             | core
 | Supported on Minion | Yes
+| Configuration Files | $OPENNMS_HOME/etc/jmx-datacollection-config.xml and $OPENNMS_HOME/etc/jmx-datacollection-config.d/*.xml
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/collectors/jmx.adoc:14]: Remove "and" and put on two separate lines?

File: docs/modules/operation/pages/performance-data-collection/collectors/prometheus.adoc [hunk 7]
```diff
@@ -14,6 +14,7 @@ This collector provides tools for parsing and mapping the metrics to the collect
 | Class Name          | `org.opennms.netmgt.collectd.prometheus.PrometheusCollector`
 | Package             | core
 | Supported on Minion | Yes
+| Configuration Files | $OPENNMS_HOME/etc/prometheus-datacollection-config.xml and $OPENNMS_HOME/etc/prometheus-datacollection.d/*.xml
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/collectors/prometheus.adoc:17]: Remove "and" and put on two separate lines?

File: docs/modules/operation/pages/performance-data-collection/collectors/wsman.adoc [hunk 8]
```diff
@@ -14,6 +14,8 @@ Windows Remote Management (WinRM) is the Microsoft implementation of WS-Manageme
 | Class Name          | `org.opennms.netmgt.collectd.WsManCollector`
 | Package             | core
 | Supported on Minion | Yes
+| Configuration Files | $OPENNMS_HOME/etc/wsman-datacollection-config.xml and $OPENNMS_HOME/etc/wsman-datacollection.d/*.xml
```
Comment (Bonrob2) [inline docs/modules/operation/pages/performance-data-collection/collectors/wsman.adoc:17]: Remove "and" and put on two separate lines?

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
A few changes.

File: docs/modules/operation/pages/performance-data-collection/collectors/wsman.adoc (hunk 8 above)
Comment (mmahacek) [reply docs/modules/operation/pages/performance-data-collection/collectors/wsman.adoc:17]:
```suggestion
| Configuration Files | $OPENNMS_HOME/etc/wsman-datacollection-config.xml +
                        $OPENNMS_HOME/etc/wsman-datacollection.d/*.xml
```

**Review by mmahacek — COMMENTED** [review]
Adding changes from other reviews

**Pull request merged**